SalesforceCommerceCloud/pwa-kit
# Add instanceType to Einstein activity body

This PR adds a new instanceType property to Einstein activities that are sent from the PWA. This will allow Reports and Dashboards to filter out activities that come from production vs activities that come from non-production environments.

In this implementation, the instanceType is inferred from the app config, where a new value `markEinsteinAnalyticsAsProduction` property has been added. Projects should change the value of `markEinsteinAnalyticsAsProduction` via [an environment specific config file](https://developer.salesforce.com/docs/commerce/pwa-kit-managed-runtime/guide/configuration-options.html#environment-specific-configurations).

Longer term, instanceType should be inferred from the `is_production` field from a Managed Runtime target. However, the mechanism for reading this field from MRT in user code currently does not exist.


Considerations:
* ~~I opted to name the property `ssrIsProduction` rather than `isProduction` in the default app config to signify that this value should be inferred from MRT. (this matches the naming convention we use for other MRT properties in the config file like ssrParameters)~~ `ssrIsProduction` has been renamed to `markEinsteinAnalyticsAsProduction`
* The default app config sets `markEinsteinAnalyticsAsProduction` to false. This makes code in the EinsteinAPI class set the instanceType to `sbx`. (In the initial implementation of Reports and Dashboards, the filter that separates production and non-production data is logic that checks for `instanceType=prd`.)


# How to Test-Drive This PR
* Start the retail react app and open devTools
* Navigate to any page
* View the Einstein activity in the network tab and verify that `instanceType=sbx` is sent as a property
* Change the `ssrIsProduction` property in default.js from false to true and reset the app
* Navigate to any page
* * View the Einstein activity in the network tab and verify that `instanceType=prd` is sent as a property
